test/build.sh

```diff
@@ -0,0 +1,137 @@
+PID=$$
+TEST_DIR=$( pwd )                   # Directory with this script
+TOP_DIR=${TEST_DIR}/..              # Top level (umbrella repo) directory
+TEST_OUTPUT=${TEST_DIR}/build_test${PID}.out
+
+build_it=0        # Set to 1 to skip build (for testing pass/fail criteria)
+#-----------------------------------------------------------------------
+# Create the output file if it doesn't exist
+#-----------------------------------------------------------------------
+if [ ! -f "$TEST_OUTPUT" ]; then
+   touch ${TEST_OUTPUT}
+fi
+
+cd ${TOP_DIR}
+
+ENV_DIR=${TOP_DIR}/env
+#-----------------------------------------------------------------------
+# Arrray of all executables built
+#-----------------------------------------------------------------------
+declare -a executables_created=( chgres_cube \
+                                 emcsfc_ice_blend \
+                                 emcsfc_snow2mdl \
+                                 filter_topo \
+                                 fregrid \
+                                 fvcom_to_FV3 \
+                                 global_cycle \
+                                 global_equiv_resol \
+                                 make_hgrid \
+                                 make_solo_mosaic \
+                                 ncep_post \
+                                 orog \
+                                 orog_gsl \
+                                 regional_esg_grid \
+                                 sfc_climo_gen \
+                                 shave \
+                                 ufs_model \
+                                 vcoord_gen )
+
+#-----------------------------------------------------------------------
+# Set up the build environment and run the build script.
+#-----------------------------------------------------------------------
+  for compiler in "${compilers[@]}"; do
+    BUILD_DIR=${TOP_DIR}/build_${compiler}
+    BIN_DIR=${TOP_DIR}/bin_${compiler}
+    EXEC_DIR=${BIN_DIR}/bin
+    if [ $build_it -eq 0 ] ; then
+      ./devbuild.sh ${machine} --compiler=${compiler} --build-dir=${BUILD_DIR} --install-dir=${BIN_DIR} \
+        --clean || fail "Build ${machine} ${compiler} FAILED"
+    fi    # End of skip build for testing
+
+  #-----------------------------------------------------------------------
+  # check for existence of executables.
+  #-----------------------------------------------------------------------
+  n_fail=0
+  for file in "${executables_created[@]}" ; do
+    exec_file=${EXEC_DIR}/${file}
+    if [ -f ${exec_file} ]; then
+      echo "SUCCEED: ${compiler} executable file ${exec_file} exists" >> ${TEST_OUTPUT}
+    else
+      echo "FAIL: ${compiler} executable file ${exec_file} does NOT exist" >> ${TEST_OUTPUT}
+      let "n_fail=n_fail+1"
+    fi
+  done
+  #-----------------------------------------------------------------------
+  # Rename build and bin dirs for next build
+  #-----------------------------------------------------------------------
```

Comment:
I'm confused by this comment block above.  What code does it apply to?

Reply:
Removed